go/mysql/conn_flaky_test.go

```diff
@@ -820,6 +820,10 @@ func (t testRun) NewConnection(c *Conn) {
 	panic("implement me")
 }
 
+func (t testRun) ConnectionReady(c *Conn) {
```

Comment:
This is just filling in all of the other internal implementations of the `Handler` interface.

When doing this, I was thinking it might be nice to introduce a stub `UnimplementedHandler` struct that all implementation can embed that provides default no-op handlers. Similar to how gRPC gives you the stub to satisfy the interface. That way an implementation isn't _require_ to implement every callback if they're not needed.

I'd be happy to follow up with an `UnimplementedHandler` struct if you also think this is valuable. It'd help wiht the boilerplate in the tests and all that get touched in this PR.